source/guide_udocker.rst

```diff
@@ -0,0 +1,138 @@
+.. highlight:: console
+
+.. author:: katakombi <katakombi@gmail.com>
+
+.. sidebar:: About
+
+  .. image:: _static/images/udocker.png
+      :align: center
+
+####################
+Qt5WebApp in UDocker
+####################
+
+UDocker is a python-based command line tool which allows to run docker containers in unprivileged user environments. Under the hood it can utilize a variety of container runtime engines.
+
+----
+
+.. note:: Since UDocker has a similar syntax as docker it is helpful to have some practical experience with docker
+
+Short Description
+=================
+
+We will utilize UDocker to pull the official ubuntu container image. 
+Then, we will install Qt5 and QtWebApp to run an example web application.
+
+Installation
+============
+
+UDocker
+-------
+
+::
+
+ [isabell@stardust ~] mkdir -p ~/.udocker
+ [isabell@stardust ~] wget https://raw.githubusercontent.com/indigo-dc/udocker/019e4d663d226aaa0caa3b173da6d1ce8264a0ab/udocker.py -O ~/.udocker/udocker.py
+ [isabell@stardust ~] chmod u+x ~/.udocker/udocker.py
+ [isabell@stardust ~] ln -sf ~/.udocker/udocker.py ~/bin/udocker
+
+Ubuntu 18:04
+------------
+
+::
+
+ [isabell@stardust ~] udocker pull ubuntu:bionic
+ [isabell@stardust ~] udocker create --name=Ubuntu_Qt5 ubuntu:bionic
+
+Prepare the installation script (Qt5,g++,make,vim)
+
+::
+
+ [isabell@stardust ~] cat << EOF > ~/install.sh
+ #!/bin/bash
+ apt-get update && apt-get -y upgrade
+ apt-mark hold dbus # dbus creates new groups upon installation which is not possible in udocker
+ apt-get install -y vim make g++ qt5-default
+ EOF
+ [isabell@stardust ~] chmod a+x ~/install.sh
+
+Execute the script as super user inside the container using a fakeroot-like environment
+
+:: 
+
+ [isabell@stardust ~] udocker run --user=root -v ~/install.sh -i -t Ubuntu_Qt5 ~/install.sh
+
+
+Setup Qt5 WebApp
+================
+
+::
+
+ [isabell@stardust ~] wget http://stefanfrings.de/qtwebapp/QtWebApp.zip -O ~/QtWebApp.zip
+ [isabell@stardust ~] cd ~ && unzip ./QtWebApp.zip
+
+Adapt port
+----------
+
+Find a free port
+
+::
+
+ [isabell@stardust ~] WWW_PORT=$(( $RANDOM % 4535 + 61000)); netstat -tulpen | grep $WWW_PORT && echo "try again"
+
+Usually this gives you a free port on first try. If you see `try again` repeat the previous command.
+Now we adapt the settings so that the external port 443 is being mapped to the local port.
+
+::
+
+ [isabell@stardust ~] cat << EOF > ~/html/.htaccess
```

Comment:
please avoid `cat << EOF >` and similar stuff, just ask the user to edit the file with an editor of their choice.